core/services/helper/main.py

```diff
@@ -45,9 +44,12 @@ class ServiceInfo(BaseModel):
 class Helper:
     LOCALSERVER_CANDIDATES = ["0.0.0.0", "::"]
 
-    @staticmethod
-    @cache
-    def detect_service(port: int) -> ServiceInfo:
+    cache: Dict[int, ServiceInfo] = {}
+
+    def detect_service(self, port: int) -> ServiceInfo:
```

Comment:
The problem with this approach is that we are not going to hot-update information of the extension if an update happens or an extension was removed.

Reply:
For tracking purposes, IIRC in our meeting patrick suggested using a `temporary_cache` here with a timeout of ~1 minute.

Reply:
I would recommend to also add a comment explaining the reason behind it, to hot reload services metadata without rebooting.